lambdas/death-validation/src/main/java/uk/gov/di/data/lep/DeathValidation.java

```diff
@@ -43,7 +43,15 @@ private GroJsonRecord validateRequest(APIGatewayProxyRequestEvent event) {
         logger.info("Validating request");
 
         try {
-            return objectMapper.readValue(event.getBody(), GroJsonRecord.class);
+            var groJsonRecord = objectMapper.readValue(event.getBody(), GroJsonRecord.class);
+            if (groJsonRecord.recordLockedDateTime() == null && groJsonRecord.recordUpdateDateTime() == null) {
+                logger.error("Record has neither recordLocked and recordUpdate dateTimes");
+                throw new MappingException("Record has neither recordLocked and recordUpdate dateTimes");
+            } else if (groJsonRecord.recordLockedDateTime() != null && groJsonRecord.recordUpdateDateTime() != null) {
+                logger.error("Record has both recordLocked and recordUpdate dateTimes");
+                throw new MappingException("Record has both recordLocked and recordUpdate dateTimes");
+            }
```

Comment:
do we definitely only want one of these?

Reply:
According to the schema, it is an either or case. We are, however, waiting for confirmation.